edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/decoupled/entity/ShippingInstruction.java

```diff
@@ -0,0 +1,64 @@
+package org.dcsa.edocumentation.domain.decoupled.entity;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.dcsa.edocumentation.domain.persistence.entity.enums.EblDocumentStatus;
+import org.dcsa.edocumentation.domain.persistence.entity.enums.TransportDocumentTypeCode;
+import org.hibernate.annotations.JdbcTypeCode;
+import org.hibernate.type.SqlTypes;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+
+@Data
+@Builder(toBuilder = true)
+@NoArgsConstructor
+@AllArgsConstructor
+@Setter(AccessLevel.PRIVATE)
+@Entity
+@Table(name = "shipping_instruction")
+public class ShippingInstruction {
```

Comment:
I recall that the ShippingInstruction used to be a domain model and having the state machine as a part of it.  I note that we are not doing that now and we have reduced it to a DAO with "no logic" in it.

To be honest, I feel the domain design (i.e., the logic was in the object) was nicer, so it is intentional, I would like to have it restored.  However, if there is a compelling reason for this change, I am open to the discussion.

Reply:
I see this happened for Booking too (apologies for not catching it at that point)

Reply:
1. Booking and ShippingInstruction was the only object that used this design, and they did it only with the state stuff, so it was actually not in-line with the rest.

2. When changing to using a jsonb blob to store the bulk part of the data it got messy to keep the entity and blob synchronized with the domain model design.

3. The domain model design version had a bug in it where when updating the state it would do that on the existing model, then lock it, then create a new. So when the historical/locked version was saved it also updated the state making the historical version have the wrong state. Decoupling the statemachine from the entity was the easiest way to solve that bug.